### ISSUE TITLE
Problems parsing SwitchExpr with Unary/BinaryOperators

### ISSUE THREAD

**Issue by Kimmmey** [javaparser/javaparser#4440]
Hi,

I have problems parsing SwitchExprs with UnaryOperators. These problems occur in the not released snapshot version. The released version 3.25.10 works correctly. So I think these problems were added, while implementing the Guards into SwitchExpr. 

Here are test cases and the corresponding output:

Test cases:
```Java
    @Test
    void testSwitchExprUnaryMinus() {
        parseStatement("int j = switch (i) {\n" +
                       "            case 0 -> 0;\n" +
                       "            default -> -1;\n" +
                       "        };");
    }

    @Test
    void testSwitchExprUnaryNot() {
        parseStatement("boolean j = switch (i) {\n" +
                       "            case 0 -> true;\n" +
                       "            default -> !false;\n" +
                       "        };");
    }
```

Output testSwitchExprUnaryMinus():
```
[ERROR] Failures: 
[ERROR]   SwitchExprTest.testSwitchExprUnaryMinus:293 [(line 3,col 21) Parse error. Found "-", expected one of  "(" "++" "--" "@" "\"\"\"" "boolean" "byte" "char" "double" "enum" "exports" "false" "float" "int" "long" "module" "new" "null" "open" "opens" "permits" "provides" "record" "requires" "sealed" "short" "strictfp" "super" "this" "throw" "to" "transitive" "true" "uses" "void" "when" "with" "yield" "{" <CHARACTER_LITERAL> <FLOATING_POINT_LITERAL> <IDENTIFIER> <INTEGER_LITERAL> <LONG_LITERAL> <STRING_LITERAL>
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:14249)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:14094)
  com.github.javaparser.GeneratedJavaParser.SwitchEntry(GeneratedJavaParser.java:6562)
[...]
```

Output testSwitchExprUnaryNot():
```
[ERROR]   SwitchExprTest.testSwitchExprUnaryNot:301 [(line 3,col 21) Parse error. Found "!", expected one of  "(" "++" "--" "@" "\"\"\"" "boolean" "byte" "char" "double" "enum" "exports" "false" "float" "int" "long" "module" "new" "null" "open" "opens" "permits" "provides" "record" "requires" "sealed" "short" "strictfp" "super" "this" "throw" "to" "transitive" "true" "uses" "void" "when" "with" "yield" "{" <CHARACTER_LITERAL> <FLOATING_POINT_LITERAL> <IDENTIFIER> <INTEGER_LITERAL> <LONG_LITERAL> <STRING_LITERAL>
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:14249)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:14094)
  com.github.javaparser.GeneratedJavaParser.SwitchEntry(GeneratedJavaParser.java:6562)
[...]
```

It seems to be a parser problem. In both cases the UnaryOperator after the `->` symbol is the problem. It's also not a problem of the `default` case, but also occurs in other cases too.

Could someone please look into it?
I am happy to help, but I am not familiar with JavaCC.

Best regards
Kim

**Comment by Kimmmey** [comment]
Apparently it is not only unary operators in a SwitchEntry, but also binary operators that cause parser errors. 

An additional test follows:

```Java
    @Test
    void testSwitchExprWithBinaryExpr() {
        parseStatement("int i = switch (1) {\n" +
                       "    case 1 -> 1;\n" +
                       "    case 2, 3 -> 1 + 2;\n" +
                       "    default -> 1;\n" +
                       "};");
    }
```

Output:
```
[ERROR]   SwitchExprTest.testSwitchExprWithBinaryExpr:309 [(line 3,col 18) Parse error. Found "+", expected one of  "%=" "&=" "*=" "++" "+=" "--" "-=" "/=" ";" "<<=" "=" ">>=" ">>>=" "^=" "|="
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:14249)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:14094)
  com.github.javaparser.GeneratedJavaParser.StatementExpression(GeneratedJavaParser.java:6228)
  com.github.javaparser.GeneratedJavaParser.SwitchEntry(GeneratedJavaParser.java:6540)
[...]
```

Best regards

**Comment by jlerbsc** [comment]
@johannescoetzee Can you take a look at this?

**Comment by johannescoetzee** [comment]
Will do! I have a few other tasks keeping be busy, but I'll hopefully get to this at the end of this week or early next

**Comment by johannescoetzee** [comment]
I introduced this bug in my first java 21 related PR, specifically [here](https://github.com/javaparser/javaparser/pull/4364/files#diff-b17099565518ace8e0d7497f2d2226da4c88452451bc1ef038fb746e58b79c24L4619-R4638) where I changed the `Expression` expected in the switch body to `StatementExpression`. I did that to fix semi-colon issues with the lexical preserving printer, but didn't consider the fact that not every expression is a statement expression. The tests I wrote all had either empty switch bodies, or print statements in the body, so I didn't cover the expression case. I'm working on a fix now!
